enhancements/insights/insights-operator-runtime.md

```diff
@@ -0,0 +1,532 @@
+
+* `/scan-containers`
+  * Inspect the all the containers running on the worker node. The executable returns a JSON payload with the extracted data organized in a tree of `namespace/pod-name/container-id/runtime-info`:
+
+```json
+{
+  "my-namespace": {
+    "my-pod-1": {
+      "cri-o://xxxx": {
+        "os-release-id": "rhel",
+        "os-release-version-id": "8.8",
+        "runtime-kind": "Java",
+        "runtime-kind-version": "17.0.8",
+        "runtime-kind-implementer": "Red Hat, Inc.",
+        "runtimes": [
+          {
+            "name": "Quarkus",
+            "version": "3.4.1",
+          }
+        ],
+      },
+      "cri-o://yyyy": {
+        "os-release-id": "rhel",
+        "os-release-version-id": "9.2",
+      }
+    },
+    "my-pod-2": {
+      [...]
+    }
+  },
+  "other-namespace": {
+    "other-pod-1": {
+      "cri-o://zzzz": {
+        [...]
+      }
+    }
+  }
+}
+```
+
+#### insights-operator-runtime Implementation
+
+Inspecting a container is achieved by running on the same worker node that the inspected container and looking at the host process table. The insights-operator-runtime enters the container’s process namespaces to extract meaningful information about the runtime stack.
+
+The extraction of data from the running container is achieved by executing “fingerprint” executables in the process `mnt` namespace to find the relevant information by reading specific files (e.g. the `Os` and `OsVersion` values are read from the `/etc/os-release` file if it is present). These fingerprints executables are self-contained executables
+
+This cannot be achieved by a program running in the OpenShift Insights Operator deployment.
+This requires an additional resource for the operator, which is a DaemonSet that would deploy a “insights-operator-runtime” image on all the cluster worker nodes.
+
+The insights-operator-runtime image requires high privileges to run (access the worker node’s host table, enter process namespaces, etc.) and needs to be configured with a specific security context constraint.
+
+![insights-operator-runtime implementation](./insights-operator-runtime-2.png "Insights-operator-runtimer Implementation")
+
+
+#### insights-operator-runtime Constraints
+
+The insights-operator-runtime is sensitive software that requires high privileges to function.
+It needs to run as privileged and access the host table of Worker Nodes to read their `/proc` tables and run as root to be able to enter in process namespaces.
+
+A specific `SecurityContextConstraints` should be added to the `openshift-insights` to deploy the insights-operator-runtime with the required permissions.
+
+In order to minimize any security risks, its code will be audited for security and its image will be as minimal as possible.
+
+The insights-operator-runtime main program is an executable written in Rust.
```

Comment:
Is there plan to rewrite it in Go? I guess in techpreview it's OK, but for GA it would be better?